go/oasis-node/cmd/debug/txsource/workload/registration.go

```diff
@@ -0,0 +1,339 @@
+package workload
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"net"
+	"path/filepath"
+	"time"
+
+	"github.com/spf13/viper"
+	"google.golang.org/grpc"
+
+	"github.com/oasislabs/oasis-core/go/common"
+	"github.com/oasislabs/oasis-core/go/common/crypto/signature"
+	memorySigner "github.com/oasislabs/oasis-core/go/common/crypto/signature/signers/memory"
+	"github.com/oasislabs/oasis-core/go/common/entity"
+	"github.com/oasislabs/oasis-core/go/common/identity"
+	"github.com/oasislabs/oasis-core/go/common/logging"
+	"github.com/oasislabs/oasis-core/go/common/node"
+	consensus "github.com/oasislabs/oasis-core/go/consensus/api"
+	"github.com/oasislabs/oasis-core/go/consensus/api/transaction"
+	cmdCommon "github.com/oasislabs/oasis-core/go/oasis-node/cmd/common"
+	registry "github.com/oasislabs/oasis-core/go/registry/api"
+	runtimeClient "github.com/oasislabs/oasis-core/go/runtime/client/api"
+)
+
+const (
+	NameRegistration = "registration"
+
+	registryNumEntities        = 10
+	registryNumNodesPerEntity  = 5
+	registryNodeMaxEpochUpdate = 5
+)
+
+var registryLogger = logging.GetLogger("cmd/txsource/workload/registry")
+
+var ns common.Namespace
```

Comment:
Maybe put this as state in the registration structure instead of making it global? This avoids potential conflicts with other workloads.

Reply:
good idea, thanks!